examples/functions/rfc3339_parse/function.tf

```diff
@@ -0,0 +1,11 @@
+terraform {
+  required_providers {
+    time = {
+      source = "hashicorp/time"
+    }
+  }
+}
```

Comment:
Heh, this is certainly a fun consequence of needing `required_providers` for provider-defined functions. Data sources and resources do not typically show this in examples, but I could see value in including it and also not including it.

I guess I'm curious if we should go the extra mile if we are including it to make the example `terraform` block more prescriptive, e.g.

```terraform
# Provider functions must include required_providers configuration.
terraform {
  required_providers {
    time = {
      source = "hashicorp/time"
      # Setting the provider version is a strongly recommended practice
      version = "..."
    }
  }
  required_version = ">= 1.8.0"
}
```

Or if we should only show the `provider::TYPE::NAME()` part for the example as shown in https://developer.hashicorp.com/terraform/plugin/framework/functions/documentation#registry-based-documentation currently? I know its technically not valid configuration on its own, but I do worry the `terraform` block could also be confusing. 🤔

Reply:
I think that's a good idea to include the comments, added in d1d88b955a57393bc0f790d759134f0924b9204b ⚡ 

My main reason for including this in the first place was two-fold:
1. As you mention, it's required for this to be valid terraform on it's own
2. All the `hashicorp/*` addressed providers will work without a `required_providers` block by default, leading me to believe that most practitioners using these utility providers may not be aware of this requirement/assume it will be handled for them.

Are those fair assumptions?